src/smriprep/workflows/anatomical.py

```diff
@@ -1344,6 +1346,65 @@ def init_anat_fit_wf(
     else:
         LOGGER.info('ANAT Stage 10: MSM-Sulc disabled')
 
+    # Stage 11: Cortical surface mask
+    if len(precomputed.get('cortex_mask', [])) < 2:
+        LOGGER.info('ANAT Stage 11: Creating cortical surface mask')
+
+        # Merge outputs into a single list
+        merge_cortex_masks = pe.Node(
+            niu.Merge(2),
+            name='merge_cortex_masks',
+        )
+        workflow.connect([(merge_cortex_masks, outputnode, [('out', 'cortex_mask')])])
+
+        for i_hemi, hemi in enumerate(['L', 'R']):
+            select_midthickness = pe.Node(
+                niu.Select(index=i_hemi),
+                name=f'select_midthickness_{hemi}',
+            )
+            select_thickness = pe.Node(
+                niu.Select(index=i_hemi),
+                name=f'select_thickness_{hemi}',
+            )
+            cortex_mask_wf = init_cortex_mask_wf(name=f'cortex_mask_wf_{hemi}')
+            cortex_mask_wf.inputs.inputnode.hemi = hemi
+
+            workflow.connect([
+                (surfaces_buffer, select_midthickness, [('midthickness', 'inlist')]),
+                (surfaces_buffer, select_thickness, [('thickness', 'inlist')]),
+                (select_midthickness, cortex_mask_wf, [('out', 'inputnode.midthickness')]),
+                (select_thickness, cortex_mask_wf, [('out', 'inputnode.thickness')]),
+            ])  # fmt:skip
+
+            # Combine the inputs into a list
+            combine_inputs = pe.Node(
+                niu.Merge(2),
+                name=f'combine_inputs_{hemi}',
+            )
+            workflow.connect([
+                (select_midthickness, combine_inputs, [('out', 'in1')]),
+                (select_thickness, combine_inputs, [('out', 'in2')]),
+            ])  # fmt:skip
+
+            ds_cortex_mask_wf = init_ds_mask_wf(
+                bids_root=bids_root,
+                output_dir=output_dir,
+                mask_type='roi',
+                name=f'ds_cortex_mask_wf_{hemi}',
+                extra_entities={'extension': '.label.gii', 'hemi': hemi},
+            )
+
+            workflow.connect([
+                (cortex_mask_wf, ds_cortex_mask_wf, [('outputnode.roi', 'inputnode.mask_file')]),
+                (combine_inputs, ds_cortex_mask_wf, [('out', 'inputnode.source_files')]),
+                (ds_cortex_mask_wf, merge_cortex_masks, [
+                    ('outputnode.mask_file', f'in{i_hemi + 1}'),
+                ]),
+            ])  # fmt:skip
```

Comment:
minor gripe: we should aim to keep the high-level workflow as simple as possible - all this wrangling should be contained within the workflow, like `init_hcp_morphometrics`.

Reply:
I think the wrangling here is because the stuff that `init_ds_surfaces_wf` does for pairs of surfaces is missing from `init_ds_mask_wf`. I would suggest seeing if we can update `init_ds_surfaces_wf` to handle whatever it lacked from masks. Or just create a hybrid `init_ds_surface_masks_wf`, since it was a special enough case to separate out from volumes.

Reply:
I went with `init_ds_surface_masks_wf`. I think the main workflow should be sufficiently clean now as well.